### PR TITLE
V1 code review

### PR DESCRIPTION
**Pull request by spanish-inquisition-4**
In current version of project we:
- configured pom file
- created .gitignore and readme file
- created script for checking code quality

### CODE REVIEW

File: checkQuality.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#! /bin/bash
+
+#create quality analisys
+mvn clean install
+mvn site
+mvn sonar:sonar
```
Comment (yakisoba0) [inline checkQuality.sh]: +1

Comment (paulstarbi) [inline checkQuality.sh]: :+1:

File: README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- One game only
+- 10x10 board
+- Fleet consists of: 4-mast ship, 2 3-mast ships, 3 2-mast ships and 4 1-mast ships.
+- Winner has ships remaining while loser has none.
+- Game messages have configurable target: console (System.out, System.err) or logs or external printer.
+- We are bi-lingual: Polish and English are fine. In future we want to add more languages: messages are to be read from a file for chosen language. Choosing the language depends on configuration variable.
+
+# Game features:
+
+- Drawing the boards for a player (fleet board has player's fleet and where opponent shot, "seen" board has where player fired and what he has shot).
```
Comment (yakisoba0) [inline README.md]: Better: Hit board keeps information about where and what player has already hit.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+- Placing the fleet - diagonal placing is disallowed, only horizontal and vertical. Humans can place ships but they can also choose to randomize placement. Ships cannot touch (no adjacent field to a ship can have a ship). Ships can be partially vertical and partially horizontal, if they have the length.
+- Firing the shot - choose a place, shoot. If you hit, you repeat the shot. You can repeat as many times as you hit.
+- Hitting the ship - hit happens when place chosen has enemy ship. Mark this part of ship as hit, ask for another shot. One can repeat the shot into already hit (or even sunken) ship, but this doesn't give the right to another shot.
+- Missing the ship - misses are marked on "seen" board. One can shoot twice in the same place if it's a miss.
```
Comment (yakisoba0) [inline README.md]: "Seen" or "Hit" as i named it before :)

File: README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+- Winner has ships remaining while loser has none.
+- Game messages have configurable target: console (System.out, System.err) or logs or external printer.
+- We are bi-lingual: Polish and English are fine. In future we want to add more languages: messages are to be read from a file for chosen language. Choosing the language depends on configuration variable.
+
+# Game features:
+
+- Drawing the boards for a player (fleet board has player's fleet and where opponent shot, "seen" board has where player fired and what he has shot).
+- Placing the fleet - diagonal placing is disallowed, only horizontal and vertical. Humans can place ships but they can also choose to randomize placement. Ships cannot touch (no adjacent field to a ship can have a ship). Ships can be partially vertical and partially horizontal, if they have the length.
+- Firing the shot - choose a place, shoot. If you hit, you repeat the shot. You can repeat as many times as you hit.
+- Hitting the ship - hit happens when place chosen has enemy ship. Mark this part of ship as hit, ask for another shot. One can repeat the shot into already hit (or even sunken) ship, but this doesn't give the right to another shot.
```
Comment (yakisoba0) [inline README.md]: Better: Hit happens when targeted field is occupied by enemy's ship.

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        <compiler.plugin.version>3.6.1</compiler.plugin.version>
+        <deploy.plugin.version>2.8.2</deploy.plugin.version>
+        <install.plugin.version>2.5.2</install.plugin.version>
+        <report.plugin.version>2.9</report.plugin.version>
+        <jar.plugin.version>3.0.2</jar.plugin.version>
+        <resources.plugin.version>3.0.2</resources.plugin.version>
+        <site.plugin.version>3.6</site.plugin.version>
+        <surefire.plugin.version>2.20</surefire.plugin.version>
+        <enforcer.plugin.version>1.4.1</enforcer.plugin.version>
+    </properties>
```
Comment (yakisoba0) [inline pom.xml]: 👍

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+                <reportSets>
+                    <reportSet>
+                        <reports>
+                            <report>report</report>
+                        </reports>
+                    </reportSet>
+                </reportSets>
+            </plugin>
+        </plugins>
+    </reporting>
```
Comment (yakisoba0) [inline pom.xml]: All necessary dependencies 👍

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+                    <execution>
+                        <id>jacoco-site</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>report</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
```
Comment (yakisoba0) [inline pom.xml]: All necessary plugins 👍

File: README.md [hunk 8]
```diff
@@ -43,4 +43,4 @@ Reports will be created in:
 
 # BONUSES
 
-- BW: additional 1h to finish the project (activity on git lecture)
\ No newline at end of file
+- BW: additional 1h to finish the project (activity on git lecture)
```
Comment (yakisoba0) [inline README.md]: Good to know 👍

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+    </build>
+</project>
```
Comment (paulstarbi) [inline pom.xml]: It looks like all works ok :+1:

File: README.md [hunk 10]
```diff
@@ -40,7 +43,3 @@ Reports will be created in:
 # TODO List
 
 [Trello playersBoard](https://trello.com/b/ONK9r1ZU/battleships)
-
-# BONUSES
-
-- BW: additional 1h to finish the project (activity on git lecture)
```
Comment (yakisoba0) [inline README.md]: Oh... Ok?

File: README.md [hunk 11]
```diff
@@ -1,3 +1,10 @@
+# For code reviewers
+
+In current version of project we:
+- configured pom file
+- created .gitignore and readme file
+- created script for checking code quality
+
```
Comment (yakisoba0) [inline README.md]: 👍

File: README.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+# About
+
+Java implementation of Battleship game (client-server architecture with simple GUI in JavaFX).<br />
```
Comment (LIttleAncientForestKami) [inline README.md]: See Markdown syntax - you don't need HTML code, like `<BR>`.

File: README.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+See game rules: https://en.wikipedia.org/wiki/Battleship_game<br />
+One additional feature is added (Nuke - see Game features)
+
+# How to run the game?
+
+It's not possible to run the game at this stage.
```
Comment (LIttleAncientForestKami) [inline README.md]: Clear-cut statement. Nice.

File: README.md (hunk 2 above)
Comment (LIttleAncientForestKami) [inline README.md]: And missed. So not hit, but shot. Shot board sounds strange though.

File: README.md (hunk 4 above)
Comment (LIttleAncientForestKami) [inline README.md]: Consistency: just above I used choose a place. Now I'm referring to chosen place. I did not use field, cause I didn't want you guys to think of Arrays through my choice of words. :-)

File: checkQuality.sh [hunk 14]
```diff
@@ -0,0 +1,9 @@
...
+
+#display results
+firefox target/site/index.html -new-window http://localhost:9000
```
Comment (LIttleAncientForestKami) [inline checkQuality.sh]: New window is a VERY nice touch, :+1:

File: pom.xml (hunk 6 above)
Comment (LIttleAncientForestKami) [inline pom.xml]: Nice POM!

File: pom.xml [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <rules>
+                        <requireMavenVersion>
+                            <version>${maven.version}</version>
+                        </requireMavenVersion>
+                        <requireJavaVersion>
+                            <version>${java.version}</version>
```
Comment (LIttleAncientForestKami) [inline pom.xml]: :+1: for this and compiler using same var.

**Pull request merged**